products-controller.js

```diff
@@ -13,9 +13,19 @@ let products = [{
 }
 ]
 
-// TODO -> end points handlers:
-
-
+// update Handler
+const updateHandler = (req,res)=>{
+   var index = products.findIndex((product) => product.code === req.params.code)
+   // check if product exist or not
+   if (index !== -1)
+    {
+        products[index] = req.body
+        res.send(products[index]) 
+    }
+    else
```

Comment:
This update method is not safe you risk losing.
it's best to add some validation before this kind of assignment, if the user of your API send a bad request containing a different code, you have essentially deleted the current product and added a new one.

I understand you are using method "PUT" but this is mostly just semantics, you need to make sure the body contains all required fields otherwise you will be losing data.